lib/veritaserum.ex

```diff
@@ -27,44 +27,69 @@ defmodule Veritaserum do
       3
   """
   @spec analyze(String.t) :: Integer.t
-  def analyze(input) do
-    input
-    |> clean
-    |> String.split
-    |> split_on_emoticons
-    |> Enum.map(&String.trim/1)
-    |> analyze_list()
-    |> Enum.reduce(0, &(&1 + &2))
-  end
 
-  defp analyze_list([head | tail]) do
-    analyze_list(tail, head, [analyze_word(head)])
+  def analyze(input, options \\ [])
+  def analyze(input, options) do
+    return = Keyword.get(options, :return, :score)
```

Comment:
Splitting this into two functions with pattern matching, we should be able to get rid of this line